java-ast/addon/src/main/java/org/jboss/windup/ast/java/ReferenceResolvingVisitor.java

```diff
@@ -193,6 +195,68 @@ private String extractDefinitionLine(String typeDeclaration)
         return typeLine;
     }
 
+    private static String extractEnumConstant(Class clazz, String line)
+    {
+        String extractedEnum = "";
+
+        // class names
+        String shortName = clazz.getSimpleName() + ".";
+        String longName = clazz.getCanonicalName() + ".";
+
+        // available enum constants
+        List constants = Arrays.asList(clazz.getEnumConstants());
+
+        List lineTokens = Arrays.asList(line.split(" "));
+        for (Iterator iterator = lineTokens.iterator(); iterator.hasNext();)
+        {
+            String object = (String) iterator.next();
+            if (object.contains(shortName))
+            {
+                int totalLen = object.length();
+                int start = object.lastIndexOf(shortName) + shortName.length();
+                String guess = object.substring(start, totalLen - 1);
+                guess = guess.replaceAll("[);.]", "");
+                Enum guessEnum = Enum.valueOf((Class<Enum>) clazz, guess);
+                if (constants.contains(guessEnum))
+                {
+                    extractedEnum = longName + guess;
+                    return extractedEnum;
+                }
+            }
+        }
+
+        return extractedEnum;
+    }
+
+    /**
+     * Extracts Enum constant from string line for a given string clazzName representing a class enum.
+     *
+     * @param clazzName - class for an Enum
+     * @param line - line with referenced Enum constant
+     * @return Fully referenced Enum constant
+     */
+    private static String extractEnumConstantFromString(String clazzName, String line)
+    {
+        Class clazz = null;
+        try
+        {
+            clazz = Class.forName(clazzName);
```

Comment:
In the case of assignment, it is possible to get it from JDT (expression.getName(), I think). I'm not sure about the other cases.